src/DynamoManipulation/TranslationGizmo.cs

```diff
@@ -395,47 +328,62 @@ public Vector GetOffset(Point newPosition, Vector viewDirection)
+            return Vector.ByTwoPoints(PointOrigin, hitPoint);
         }
 
         /// <summary>
         /// Gets drawables to render this Gizmo
         /// </summary>
-        /// <param name="factory">Render package factory</param>
         /// <returns>List of render package</returns>
-        public IEnumerable<IRenderPackage> GetDrawables(IRenderPackageFactory factory)
+        public override IEnumerable<IRenderPackage> GetDrawables()
         {
             List<IRenderPackage> drawables = new List<IRenderPackage>();
-            foreach (var axis in axes)
+            foreach (Vector axis in axes)
             {
-                IRenderPackage package = factory.CreateRenderPackage();
+                IRenderPackage package = RenderPackageFactory.CreateRenderPackage();
                 DrawAxis(ref package, axis);
                 drawables.Add(package);
             }
 
             var p = Planes.xyPlane;
-            foreach (var plane in planes)
+            foreach (Plane plane in planes)
             {
-                IRenderPackage package = factory.CreateRenderPackage();
+                IRenderPackage package = RenderPackageFactory.CreateRenderPackage();
                 DrawPlane(ref package, plane, p++);
                 drawables.Add(package);
             }
-            drawables.AddRange(GetDrawablesForTransientGeometry(factory));
+            drawables.AddRange(GetDrawablesForTransientGraphics());
 
             return drawables;
         }
 
-        public void HighlightGizmo(IWatch3DViewModel backgroundPreviewViewModel, 
-            IRenderPackageFactory factory)
+        public override void HighlightGizmo()
+        {
+            var drawables = GetDrawablesForTransientGraphics();
+            BackgroundPreviewViewModel.AddGeometryForRenderPackages(drawables);
+        }
+
+        public override void UpdateGizmoGraphics()
         {
-            var drawables = GetDrawablesForTransientGeometry(factory);
-            backgroundPreviewViewModel.AddGeometryForRenderPackages(drawables);
+            // Update gizmo geometry wrt to current Origin
+            var newPlanes = planes.Select(
+                plane => Plane.ByOriginXAxisYAxis(Origin, plane.XAxis, plane.YAxis)).ToList();
+            planes.Clear();
+            planes.AddRange(newPlanes);
+
+            DeleteTransientGraphics();
         }
 
-        public void UnhighlightGizmo(IWatch3DViewModel backgroundPreviewViewModel)
+        public override void UnhighlightGizmo()
         {
             // Delete all transient geometry used to highlight gizmo
-            backgroundPreviewViewModel.DeleteGeometryForIdentifier(RenderDescriptions.AxisLine);
+            DeleteTransientGraphics();
+        }
+
+        public override void DeleteTransientGraphics()
+        {
+            var identifier = string.Format("{0}_{1}", RenderDescriptions.AxisLine, Name);
+            BackgroundPreviewViewModel.DeleteGeometryForIdentifier(identifier);
```

Comment:
this fixes the issue with [highlighting not working with all active gizmos in the view] (http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-9164)